### DISCUSSION TITLE
Custom path format for some imports

### DISCUSSION THREAD

**Discussion by trapd00r** [beetbox/beets#4012] [Q&A] [answered]
At times I want to place some albums or singletons in a directory that doesn't correspond to the main artist. A few examples:

```
# I want this in the Cardigans directory
Tom Jones and The Cardigans │1999│ Burning Down the House [Single, CD, MP3]

# I want these in the 1200 Micrograms directory
Johann Bley - Stranded (1200 Micrograms Rmx).mp3
Message In A Bottle (1200 Micrograms RMX).mp3

```
So far, I've done things like this to achieve this:

```yaml
tag:eminem: E/Eminem/....
```
and a 
```bash
beet import --set tag=eminem Artist_that_is_not_Eminem*
```
But it's not feasible to do it like this every time I find a case where I want to place things differently. 

I do **not** want to use the rewrite plugin but I'd rather have a way to specify a directory at import time, somehow. 

How would **you** do it?

**Comment by sampsyo** [comment, marked as answer]
Hmm… how about using a flexible field to contain the directory name you want? That is, you could set a `myartist` field and, when it’s nonempty, use `$myartist` in the path format.

**Reply by trapd00r** [reply]
Excellent! I was able to modify a function I already use: https://github.com/trapd00r/configs/blob/9b39c0bf8e3818bb07b2bdf2980b744573bd4ae3/beets/config.yaml#L550

And then:
`    
    default:                     %bucket{$first_artist, alpha}/%titlecase_proper{$first_artist}/%titlecase_proper{$albumartist} ...

`

Now I can do things like

```
 beet import -s --set myartist='1200 Micrograms' *(.) 
```
which results in
```
/mnt/music8/+TAGGED/0-9/1200 Micrograms/+tracks/2015 Johann Bley - Stranded (1200 Micrograms Remix).mp3
```

Thank you Adrian!
